registry/docker/testing/server.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package testing provides a fake implementation of the registry API.
+
+package testing
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/mux"
+)
+
+type Repository struct {
```

Comment:
I think we could use a different name here, like `ImageRepository`, to avoid confusion with the `repository` package.